app/(login)/LoginScreen.tsx

```diff
@@ -0,0 +1,108 @@
+import styles from './styles'
+import { Text, View, } from '../../components/Themed';
+import { TextInput } from 'react-native-gesture-handler';
+import { ImageBackground, Pressable } from 'react-native';
+import { Component } from 'react';
+import { Link, router } from 'expo-router';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import { LogoAndName } from '../../components/LogoAndName';
+
+type Props = {
+  text: any
+}
+
+type State = {
+  phoneNumber: string
+  password: string,
+  wrongCredentials: boolean,  
+  hidePassword: boolean,
+  authing: boolean
+}
+
+export default class loginScreen extends Component<Props, State> {
```

Comment:
React does not recommend defining components as classes, but instead as functions per: https://react.dev/reference/react/Component
They have a guide for migration here: https://react.dev/reference/react/Component#alternatives